app/models/solicitation.rb

```diff
@@ -88,6 +88,10 @@ def options=(options_hash) # Support for old-style solicitation form. TODO: Remo
     self.landing_options_slugs = options_hash.select{ |_, v| v.to_bool }.keys
   end
 
+  def self.all_past_landing_options_slugs
+    self.distinct.pluck("unnest(#{:landing_options_slugs})").count
```

Comment:
elle sert à quoi cette méthode ?

Reply:
(je sais pas exactement de quelle méthode tu parles alors:)
* `Solicitation.all_past_landing_options_slugs` (méthode de classe) permet d’obtenir toutes les options créées un jour dans le passées. C’est utilisé pour l’extrait csv, pour créer une colonne par option.
😅 Je réalise que le `.count` à la fin est de trop. Blame the copier-coller.

* `unnest` sert à aplatir l’array postgresql. Ça revient au même que `Solicitation.pluck(:landing_options_slugs).flatten.uniq`, en restant en DB. Peut-être que la version ruby serait plus claire.

(D’ailleurs, j’ai pas indexé l’array, donc je suis pas sûr que ça soit très optimisé même en DB; sur ma machine ça n’avait pas l’air de poser de problème avec les données de prod anonymisées.)

Bref, j’enlève le .count et je mets des commentaires.

Reply:
Merci, en fait l'explication des deux méthodes me sont utiles